src/toeplitzhankel.jl

```diff
@@ -0,0 +1,136 @@
+function partialchol(H::Hankel)
+    # Assumes positive definite
+    σ=Array(eltype(H),0)
+    n=size(H,1)
+    C=Array(eltype(H),n,n)
+    v=[H[:,1];vec(H[end,2:end])]
+    d=diag(H)
+
+    @assert length(v) ≥ 2n-1
+
+    tol=1E-14
+    for k=1:n
+        mx,idx=findmax(d)
+        if mx ≤ tol
+            break
+        end
+        push!(σ,1/mx)
+        @simd for p=1:n
+            @inbounds C[p,k]= v[p+idx-1]             # H[p,idx]
+        end
+        for j=1:k-1,p=1:n
+            @inbounds C[p,k]-=C[p,j]*C[idx,j]*σ[j]
+        end
+        @simd for p=1:n
+            @inbounds d[p]-=C[p,k]^2/mx
+        end
+    end
+    for k=1:length(σ),p=1:n
+        @inbounds C[p,k]*=sqrt(σ[k])
+    end
+    C[:,1:length(σ)]
+end
+
+function toeplitzcholmult(T,C,v)
+    ret=C[:,1].*(T*(C[:,1].*v))
+    for j=2:size(C,2)
+        ret+=C[:,j].*(T*(C[:,j].*v))
+    end
+    ret
+end
+
+
+# Plan a multiply by DL*(T.*H)*DR
+immutable ToeplitzHankelPlan{TT}
+    T::TriangularToeplitz{TT}
+    C::Matrix{TT}   # A cholesky factorization of H: H=CC'
+    DL::Vector{TT}
+    DR::Vector{TT}
+
+    ToeplitzHankelPlan(T,C,DL,DR)=new(T,C,DL,DR)
+end
+
+
+function ToeplitzHankelPlan(T::TriangularToeplitz,C::Matrix,DL::AbstractVector,DR::AbstractVector)
+    TT=promote_type(eltype(T),eltype(C),eltype(DL),eltype(DR))
+    ToeplitzHankelPlan{TT}(T,C,collect(TT,DL),collect(TT,DR))
+end
+ToeplitzHankelPlan(T::TriangularToeplitz,C::Matrix) =
+    ToeplitzHankelPlan(T,C,ones(size(T,1)),ones(size(T,2)))
+ToeplitzHankelPlan(T::TriangularToeplitz,H::Hankel,D...) =
+    ToeplitzHankelPlan(T,partialchol(H),D...)
+
+*(P::ToeplitzHankelPlan,v::Vector)=P.DL.*toeplitzcholmult(P.T,P.C,P.DR.*v)
+
+
+
+# Legendre transforms
+
+
+Λ{T}(::Type{T},z)=z<5?gamma(z+one(T)/2)/gamma(z+one(T)):exp(lgamma(z+one(T)/2)-lgamma(z+one(T)))
```

Comment:
This may have been obscured by my poor documentation, but the function `Cx` in `specialfunctions.jl` is this `Λ`. Consider extending it rather than having some duplication? Equivalent functionality can be obtained with
```julia
Λ = Cx
```

The asymptotic series specialization for `Float64` has relative error less than machine epsilon, is computed in O(1) time, and is only marginally slower. From what I understand, this is not a dominant cost in the computation, so might as well get the connection coefficients as accurate as possible.
```julia
julia> using FastTransforms, PyPlot

julia> x = 0:0.5:10_000;

julia> @time τvals = FastTransforms.Cx(x);
  0.000895 seconds (6 allocations: 156.484 KB)

julia> @time Λvals = Λ(Float64,x);
  0.000501 seconds (6 allocations: 156.484 KB)

julia> tvals = FastTransforms.Cx(map(BigFloat,x));

julia> err1 = abs(map(Float64,τvals./tvals-1));

julia> err2 = abs(map(Float64,Λvals./tvals-1));

julia> semilogy(x,err1,".r",x,err2,".g",[1.0,10000],eps()*ones(2),"-k");xlabel("\$ x\$");ylabel("Relative Error")
PyObject <matplotlib.text.Text object at 0x320136890>

julia> legend(["Asymptotic","Recurrence","Machine epsilon"],loc="bottom right")
PyObject <matplotlib.legend.Legend object at 0x320185790>

julia> savefig("asyrec.png")
```
![asyrec](https://cloud.githubusercontent.com/assets/8176037/14936406/e3c8f0f6-0ee3-11e6-83c8-f9cc9d7c8c52.png)